Content.Client/GameObjects/Components/Mobs/ClientActionsComponent.cs

```diff
@@ -0,0 +1,981 @@
+        private RichTextLabel _actionName;
+        private RichTextLabel _actionDescription;
+        private RichTextLabel _actionCooldown;
+        private RichTextLabel _actionRequirements;
+        private bool _tooltipReady;
+        private ActionSlot _showingTooltipFor;
+        private List<ItemSlotButton> _highlightingItemSlots = new List<ItemSlotButton>();
+        // so we don't call it every frame and only update the text each second that ticks
+        private int _tooltipCooldownSecs = -1;
+        // tracks the action slot we are currently selecting a target for
+        private ActionSlot _selectingTargetFor;
+
+        private readonly ActionAssignments _assignments = new ActionAssignments(Hotbars, Slots);
+
+        /// <summary>
+        /// Allows calculating if we need to act due to this component being controlled by the current mob
+        /// TODO: should be revisited after space-wizards/RobustToolbox#1255
+        /// </summary>
+        [ViewVariables]
+        private bool CurrentlyControlled => _playerManager.LocalPlayer != null && _playerManager.LocalPlayer.ControlledEntity == Owner;
+
+        // index of currently displayed hotbar
+        private byte _selectedHotbar = 0;
+
+        protected override void Shutdown()
+        {
+            base.Shutdown();
+            PlayerDetached();
+        }
+
+        public override void HandleMessage(ComponentMessage message, IComponent component)
+        {
+            base.HandleMessage(message, component);
+            switch (message)
+            {
+                case PlayerAttachedMsg _:
+                    PlayerAttached();
+                    break;
+                case PlayerDetachedMsg _:
+                    PlayerDetached();
+                    break;
+            }
+        }
+
+        public override void HandleComponentState(ComponentState curState, ComponentState nextState)
+        {
+            base.HandleComponentState(curState, nextState);
+
+            if (!(curState is ActionComponentState))
+            {
+                return;
+            }
+
+            UpdateUI();
+        }
+
+        private void PlayerAttached()
+        {
+            if (!CurrentlyControlled || _ui != null)
+            {
+                return;
+            }
+
+            _ui = new ActionsUI(ActionOnOnShowTooltip, ActionOnOnHideTooltip, OnActionPress,
+                OnActionSlotDragDrop, OnMouseEnteredAction, OnMouseExitedAction,
+                NextHotbar,
+                PreviousHotbar, HandleOpenActionMenu);
+            _menu = new ActionMenu(ActionOnOnShowTooltip, ActionOnOnHideTooltip, this, ActionMenuItemSelected,
+                ActionMenuItemDragDropped);
+
+            var uiManager = IoCManager.Resolve<IUserInterfaceManager>();
+            uiManager.StateRoot.AddChild(_ui);
+
+            _inventoryComponent = Owner.GetComponent<ClientInventoryComponent>();
+            _handsComponent = Owner.GetComponent<HandsComponent>();
```

Comment:
This will error if these two components don't exist. Replace it with a TryGet, or better yet change `_inventoryComponent` and `_handsComponent` to use component dependencies.